audio/out/internal.h

```diff
@@ -201,7 +201,7 @@ struct ao_driver {
 
 // These functions can be called by AOs.
 
-int ao_read_data(struct ao *ao, void **data, int samples, int64_t out_time_ns);
+int ao_read_data(struct ao *ao, void **data, int samples, int64_t out_time_ns, bool *eof, bool pad_silence, bool blocking);
```

Comment:
`pad_silence` and `blocking` always have the same value.
Can't they be unified?

Reply:
> `pad_silence` and `blocking` always have the same value.

Not always. For example, in my following changes to `ao_avfoundation` (not pushed yet), I've made `pad_silence = false` but `blocking = true`; `ao_avfoundation` can tolerate both blocking and partial data.

```diff
diff --git a/audio/out/ao_avfoundation.m b/audio/out/ao_avfoundation.m
index 971eccf4db..8eaf78d3ad 100644
--- a/audio/out/ao_avfoundation.m
+++ b/audio/out/ao_avfoundation.m
@@ -77,7 +77,7 @@ static void feed(struct ao *ao)
     int64_t end_time_av = MPMAX(p->end_time_av, cur_time_av);
     int64_t time_delta = CMTimeGetNanoseconds(CMTimeMake(request_sample_count, samplerate));
     bool eof;
-    int real_sample_count = ao_read_data(ao, data, request_sample_count, end_time_av - cur_time_av + cur_time_mp + time_delta, &eof, false, false);
+    int real_sample_count = ao_read_data(ao, data, request_sample_count, end_time_av - cur_time_av + cur_time_mp + time_delta, &eof, false, true);
     if (eof) {
         [p->renderer stopRequestingMediaData];
     }
```

Reply:
If they are going to differ, sure both are needed.
Thanks for the clarification.